addons/pos_restaurant/static/src/js/floors.js

```diff
@@ -784,13 +784,12 @@ models.PosModel = models.PosModel.extend({
             this.set_order(null);
         } else if (this.order_to_transfer_to_different_table) {
             this.order_to_transfer_to_different_table.table = table;
+            this.db.save_unpaid_order(this.order_to_transfer_to_different_table);
```

Comment:
Call `save_to_db()` of `this.order_to_transfer_to_different_table` instead. That's what we use in the rest of the POS. By directly calling `save_unpaid_order()` you bypass the posmodel `init_locked` lock.